### PR TITLE
Preview in Convert && More Widgets

### PR DESCRIPTION
**Pull request by trindadedev13**
[Example Video](https://jumpshare.com/s/37dzCRdv0KYm2uJwDcvP)

main issues:
You need to re-open fragment to update type.

also added MaterialSwitch preview, because i saw much people use it, I also plan to add preview for more views in future.

### CODE REVIEW

**Comment by aikrq** [comment]
so many force pushes :skull:

**Comment by trindadedev13** [comment]
> so many force pushes 💀

lol

**Comment by hasrat-ali** [comment]
That's why local branch is good for testing.
Don't push untill it's almost ready

**Comment by trindadedev13** [comment]
> That's why local branch is good for testing. Don't push untill it's almost ready

but for test, i need to push

**Comment by trindadedev13** [comment]
most force-pushes are reverts of commits, or updating the branch with upstream

**Comment by PranavPurwar** [comment]
What about opening PR *after* you've tested everything?

**Comment by trindadedev13** [comment]
> What about opening PR _after_ you've tested everything?

Well, I opened the PR in case anyone who knows more about viewPane and wants to help me with the issue I mentioned

File: app/src/main/java/com/besome/sketch/editor/view/ViewPane.java [hunk 1]
```diff
@@ -419,122 +435,150 @@ private void updateItemView(View view, ViewBean viewBean) {
...
-            }
-            if (classInfo.b("CircleImageView")) {
-                updateCircleImageView((ItemCircleImageView) view, injectHandler);
-            } else {
-                ((ImageView) view).setScaleType(ImageView.ScaleType.valueOf(viewBean.image.scaleType));
             }
         }
         if (classInfo.a("CompoundButton")) {
-            ((CompoundButton) view).setChecked(viewBean.checked != 0);
+            if (view instanceof CompoundButton) {
```
Comment (PranavPurwar) [inline app/src/main/java/com/besome/sketch/editor/view/ViewPane.java:492]: are these extra checks necessary? I think the views would already be of proper types

Comment (trindadedev13) [reply app/src/main/java/com/besome/sketch/editor/view/ViewPane.java:492]: well that fixed some  crashs related with cannot be cast

Comment (elfilibusterismo) [reply app/src/main/java/com/besome/sketch/editor/view/ViewPane.java:492]: Make it clear, crash that caused by changing the type of viewbean

**Comment by elfilibusterismo** [comment]
> > What about opening PR _after_ you've tested everything?
> 
> Well, I opened the PR in case anyone who knows more about viewPane and wants to help me with the issue I mentioned

If you want to detect the changes of convert add another variable in ViewBean something like `preConvert` now remove the unnecessary check and do the checks in `ViewPane#g(ViewBean)`(not sure) , then add check where convert not equal preConvert

**Comment by hasrat-ali** [comment]
Is there any use for this in reality?

**Comment by elfilibusterismo** [comment]
> Is there any use for this in reality?

Yeah imo too, it doesn't really necessary now

**Comment by elfilibusterismo** [comment]
> > Since we have `Direct XML editor` the layouts
> 
> Well if we think like this, Convert itself is no longer necessary

Not really, since Convert still allows custom ones. Your changes just seem a bit unnecessary now.

**Pull request closed without merging**